### ISSUE TITLE
Failure to infer type when `type(None)` used in `isinstance` check

### ISSUE THREAD

**Issue by todd-dembrey** [microsoft/pyright#1762]
**Describe the bug**
`isinstance` checks in mypy with `type(None)` correctly infer the updated type of the variable however in pyright the type of the variable remains unchanged. 

In my code there is the following assertion which works in mypy but fails in pyright.

```python
value: Optional[Union[str, int]] 
assert isinstance(value, (int, type(None)))
reveal_type(value) # Optional[Union[str, int]] where Optional[int] expected
```

I have narrowed the error down to the use of `type(None)` within the `isinstance` check, see below.

**To Reproduce**
```python
from typing import Optional
def bad(a: Optional[str]) -> None:
    assert isinstance(a, type(None))
    return a  # Expression of type "str | None" cannot be assigned to return type "None

def good(a: Optional[str]) -> None:
    assert a is None
    return a
```

**Expected behavior**
Parity with mypy

**Additional context**
pyright version 1.1.130
mypy version 0.812

**Comment by erictraut** [comment]
Thanks for reporting. 

Support for `type(None)` within `isinstance` type narrowing will be included in the next release of pyright.

**Comment by todd-dembrey** [comment]
Brilliant, thank you.

**Comment by erictraut** [comment]
This is included in pyright 1.1.131, which I just published. It will also be included in the next release of pylance.
